src/crypto/equihash.cpp

```diff
@@ -458,6 +472,7 @@ std::set<std::vector<eh_index>> Equihash<N,K>::OptimisedSolve(const eh_HashState
     std::set<std::vector<eh_index>> solns;
     eh_index recreate_size { UntruncateIndex(1, 0, CollisionBitLength + 1) };
     int invalidCount = 0;
```

Comment:
If these initialisations were hoisted above the preceding loop, would it be ok to use ``if (cancelled(FinalColliding)) goto cancel solver;`` above, thus simplifying the test?